### ISSUE TITLE
Closing Registration Window Loses Focus

### ISSUE THREAD

**Issue by chinakov** [audacity/audacity#9097]
### Bug description

If the User opens the new registration window via the welcome window by pressing 'continue', and then closes it, the application will lose focus and keyboard users won't be able to close or re-enter the window. 

### Steps to reproduce

1. Open Audaicty
2. Observe welcome dialog window
3. Press 'Continue' on audio.com slidee
4. Close the regsitration window
5. Try to tab around
6. Observe result

### Expected behavior

The focus should be switched back to the welcome window after registration dialog has been closed. 

### Actual behavior

The focus is not switched back to the welcome window after registration dialog has been closed. 

### Audacity Version

ba024db89189c857f3455bde1dbfbe5fd17bc11c

### Operating system

OS independent

### Additional context

https://github.com/user-attachments/assets/d375a4a9-cbbf-408f-aa95-743eab8aa86c

**Comment by kryksyh** [comment]
This is also the case for "Share Audio" and "Save to Cloud" dialogs.

**Comment by DavidBailes** [comment]
> This is also the case for "Share Audio" and "Save to Cloud" dialogs.

@kryksyh . I couldn't reproduce the problem on Windows. Could you give the exact steps to reproduce for one of these?

**Comment by kryksyh** [comment]
@DavidBailes , for both "Share Audio" and "Save to Cloud" dialogs the behavior is the same:

- Launch Audacity
- Log out from Audio.com
- Create a clip
- Press "Share Audio" toolbar button
- Enter the track name and press Continue
- "Create Cloud Account" shows up
- Press "Esc" to close the dialog
- Observe that "Share Audio" dialog is not focused

**Comment by kryksyh** [comment]
It is even worse, actually. After I log in via the browser and log out, when I press "Esc" to close the Sign-In dialog, my browser receives the focus and jumps to the front.

**Comment by DavidBailes** [comment]
> It is even worse, actually. After I log in via the browser and log out, when I press "Esc" to close the Sign-In dialog, my browser receives the focus and jumps to the front.

Yes, Audacity loses the focus completely! Thanks for the steps to reproduce.

**Comment by chinakov** [comment]
Thanks, this has been fixed.
